addons/x_lib/functions/config/fnc_getArtyMagazineType.sqf

```diff
@@ -48,110 +44,69 @@ private _ord="";
-                        if (_atmine) then {
-                         _ords = [_ords, _x] call ALIVE_fnc_push;
-                         };
-
-                    } forEach _mags;
-                };
-        case "ROCKETS": {
-                    {
-                         private _rockets = ["rockets", _x] call BIS_fnc_inString;
-                        if (_rockets) then {
-                             _ords = [_ords, _x] call ALIVE_fnc_push;
-                         };
-
-                    } forEach _mags;
-                };
-
-            case "ILLUM": {
-                    {
-                         private _illum = ["illum", _x] call BIS_fnc_inString;
-                     if (!_illum) then {
-                        _illum = ["flare", _x] call BIS_fnc_inString;
-                        };
-                        if (_illum) then {
-                             _ords = [_ords, _x] call ALIVE_fnc_push;
-                         };
-
-                    } forEach _mags;
-                };
-        case default {
-                    _ord="";
-                };
+        {
+            private _smoke = ["Mo_smoke", _x] call BIS_fnc_inString;
+            if (!_smoke) then { _smoke = ["smoke", _x] call BIS_fnc_inString; };
+            if (_smoke) then { _ords pushBack _x; };
+        } count _mags;
+    };
+    case "SADARM": {
+        {
+            private _guided = ["Mo_guided", _x] call BIS_fnc_inString;
+            if (_guided) then { _ords pushBack _x; };
+        } count _mags;
+    };
+    case "CLUSTER": {
+        {
+            private _cluster = ["Mo_Cluster", _x] call BIS_fnc_inString;
+            if (_cluster) then { _ords pushBack _x; };
+        } count _mags;
+    };
+    case "LASER": {
+        {
+            private _lg = ["Mo_LG", _x] call BIS_fnc_inString;
+            if (!_lg) then { _lg = ["laser", _x] call BIS_fnc_inString; };
+            if (_lg) then { _ords pushBack _x; };
+        } count _mags;
+    };
+    case "MINE": {
+        {
+            private _mine = ["Mo_mine", _x] call BIS_fnc_inString;
+            if (_mine) then { _ords pushBack _x; };
+        } count _mags;
+    };
+    case "AT MINE": {
+        {
+            private _atmine = ["Mo_AT_mine", _x] call BIS_fnc_inString;
+            if (_atmine) then { _ords pushBack _x; };
+        } count _mags;
+    };
+    case "ROCKETS": {
+        {
+            private _rockets = ["rockets", _x] call BIS_fnc_inString;
+            if (_rockets) then { _ords pushBack _x; };
+        } count _mags;
+    };
+
+    case "ILLUM": {
+        {
+            private _illum = ["illum", _x] call BIS_fnc_inString;
+            if (!_illum) then { _illum = ["flare", _x] call BIS_fnc_inString; };
+            if (_illum) then { _ords pushBack _x; };
+        } count _mags;
+    };
+    case default { _ord = ""; };
 };
-;
-_ord = _ords select 0;
 
-
-
-_ord;
+_ord = _ords select 0;
+_ord
 
 
 
```

Comment:
Too much lines to comment on ^^.

Use `forEach` instead of `count` and `if` inner block(s) should be on their own line.